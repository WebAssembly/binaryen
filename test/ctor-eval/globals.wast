(module
  (global $g1 (mut i32) (i32.const 10))
  (global $g2 (mut i32) (i32.const 20))
  (func $test1 (export "test1")
    (global.set $g1 (i32.const 30))
    (global.set $g2 (i32.const 40))
    (global.set $g1 (i32.const 50)) ;; this overrides the previous 30
  )
  (func $keepalive (export "keepalive") (result i32)
    ;; Keep the globals alive so we can see their values.
    (i32.add
      (global.get $g1)
      (global.get $g2)
    )
  )
)
