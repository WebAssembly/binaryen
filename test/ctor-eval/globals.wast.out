(module
 (type $none_=>_i32 (func (result i32)))
 (global $g1 (mut i32) (i32.const 50))
 (global $g2 (mut i32) (i32.const 40))
 (export "keepalive" (func $keepalive))
 (func $keepalive (result i32)
  (i32.add
   (global.get $g1)
   (global.get $g2)
  )
 )
)
